### ISSUE TITLE
Initial Challenge Setup - Token Voting

### ISSUE THREAD

**Issue by escottalexander** [BuidlGuidl/eth-tech-tree-challenges-old#2]
The setup is like this; `eth-tech-tree` is a CLI for downloading, submitting and managing which challenges a user has completed. The repo that houses all the challenges in called `eth-tech-tree-challenges`. Each challenge is a new branch on that repo.

Follow the example left by the [Token Wrapper WETH challenge](https://github.com/BuidlGuidl/eth-tech-tree-challenges/tree/token-wrapper-weth) and see how that challenge is configured in the [challenges.json file in the eth-tech-tree](https://github.com/BuidlGuidl/eth-tech-tree/blob/main/challenges.json) repo.

``` json
{
    "type": "challenge",
    "level": 1,
    "name": "token-wrapper-weth", // Branch name
    "label": "Token Wrapper - WETH",
    "repo": "https://github.com/escottalexander/scaffold-eth-2",
    "testHash":"cdef2b9c0051c45a8ce2c1ec14a7446419496a018a6dc114735cce9de3a4f249",
    "testFileName":"WrappedETH.t.sol",
    "tags": ["DeFi"]
  },
```
Here are the steps for adding a new challenge:
### Creating a new challenge
1. Fork and download the [eth-tech-tree-challenges](https://github.com/BuidlGuidl/eth-tech-tree-challenges) repo
2. Create a new branch based on the main branch (which is set up to be a template for any challenge) with the name of the challenge e.g. 'token-voting'
3. Write out a solution to the challenge in solidity using best industry standards as generically as possible by editing Challenge.sol and changing it's name to match the challenge
4. Write tests that cover the most common pitfalls and guarantee the contract operates as expected (Challenge.t.sol) - change it's name to match the contract
5. Update README in your challenge to give the user clarity about what they must accomplish and also any helpful resources (see the token-wrapper-weth challenge for an example)
6. Commit and push your changes to your remote repo, creating a new branch on your remote `git push`
7. Submit a PR to the main eth-tech-tree-challenges repo pointed to the branch that matches your challenge name e.g. 'token-voting'

### Adding the new challenge to eth-tech-tree CLI app
8. Edit challenges.json (in eth-tech-tree) to have proper values for each property        
    - name: should match your branch name    
    - repo: should be your repository
    - testName: should be the name of the test file
    - testHash: should be the hash of the test file. Use console.log to get this when taking the "submit challenge" action on your challenge when running eth-tech-tree CLI
9. Test downloading and submitting the challenge using the eth-tech-tree CLI `yarn build && yarn cli`
10. Commit and push changes to the eth-tech-tree in your own fork and branch called 'adding-[challenge branch name]' and submit a PR to the main repo 🎉

Later, in another task, we will erase your specific solution code from the git history for your new branch in eth-tech-tree-challenges.

------------------------------------------------------------------------------------------------

## Token Voting challenge specifics

This challenge entails creating a voting contract that has the specific purpose of receiving votes for/against a specific proposal. This will not include creating proposals so just create the contract with an assumed notion people will vote on. 
A user will vote only once on the proposal and their votes weight is equal to the amount of tokens they hold. Note: they don't need to actually transfer their tokens to the voting contract, just signal their vote. The contract should have a set voting deadline at which the majority wins and upon calling the contract it returns the result. Feel free to ask any questions or express any ideas that will help the end user learn through this challenge.

It already has an object in challenges.json which has the name field 'token-voting'.
